api/v1beta1/hetznerbaremetalmachine_webhook.go

```diff
@@ -50,71 +43,17 @@ var _ webhook.Validator = &HetznerBareMetalMachine{}
 
 // ValidateCreate implements webhook.Validator so a webhook will be registered for the type.
 func (bmMachine *HetznerBareMetalMachine) ValidateCreate() (admission.Warnings, error) {
-	var allErrs field.ErrorList
-
-	if bmMachine.Spec.SSHSpec.PortAfterCloudInit == 0 {
```

Comment:
don't you have to keep this code block? you also added it to the other webhook for the template object